BUILD.bazel

```diff
@@ -234,27 +233,16 @@ alias(
     visibility = ["//visibility:public"],
 )
 
-cc_binary(
```

Comment:
Looks like there's still a bunch of test failures related to the alias/cc_binary switch.  There are a pretty large number of places where we assume protoc and protoc_static get dumped directly into bazel-bin, and this change moves the binaries somewhere else.

I agree with the direction, and it would be nice to keep as much as possible in //src/google/protobuf/compiler, but I think the cc_binary targets need to live here for now (to avoid a major refactor)